docker/debian/Dockerfile

```diff
@@ -118,3 +145,21 @@ WORKDIR /home/${NONROOT_USER}
 
 # Create a default Conan profile.
 RUN conan profile detect
+# Fix the C++ dialect.
+RUN sed -i -e 's|^compiler\.cppstd=.*$|compiler.cppstd=20|' ~/.conan2/profiles/default
+# To ensure compatibility with a range of Clang compilers, we must add extra
+# flags that apply to certain versions of Clang.
+RUN <<EOF
+if [[ $(clang-${CLANG_VERSION} --version | head -1 | grep -Po 'version \K[0-9]{2}') -ge 20 ]]; then
+    cat >>~/.conan2/profiles/default <<EOT
+tools.build:cxxflags=['-Wno-missing-template-arg-list-after-template-kw', '-Wno-deprecated-declarations']
```

Comment:
I do not like that we are adding the '-Wno-missing-template-arg-list-after-template-kw' workaround because the problem is not in the toolset, but in a dependency (old version of `grpc`, to be specific). But I guess as a temporary workaround we need to go ahead with this.